Tests/ApodiniTests/OpenAPITests/OpenAPIComponentsObjectBuilderTests.swift

```diff
@@ -0,0 +1,127 @@
+//
+//  Created by Lorena Schlesinger on 29.11.20.
+//
+
+import XCTest
+import Foundation
+import OpenAPIKit
+import Runtime
+import NIO
+@testable import Apodini
+
+final class OpenAPIComponentsObjectBuilderTests: XCTestCase {
+    let someString = "Some String"
+    let someInt = 42
+    let someDouble = 42.0
+    let someBool = true
+    let someDict = ["someKey": "someValue"]
+    let someArray = [1, 2, 3]
+    let someEventLoop: EventLoopFuture<Int>? = nil
+    let someEither: Either<Int, String> = .init("someString")
+    let someOptional: String? = nil
+    let someOptionalUUID: UUID? = nil
+
+    struct SomeStruct: Encodable {
+        var id: UUID?
+        var someProp = 4
+    }
+
+    struct GenericStruct<T>: Encodable {
+        var list: [T]
+        var listLength: Int
+
+        func encode(to encoder: Encoder) throws {
```

Comment:
`struct GenericStruct<T>: Encodable where T: Encodable { ... }` should do the trick.

Reply:
done in 6974958